### PR TITLE
add basePath to GenApiClient

### LINKED ISSUE #1: Base path for ApiClient
Would be nice to be able to do: 

```
/// Example showing how to define an [ApiClient]
@GenApiClient("users")
class UserApi extends _$UserApiClient implements ApiClient {
  final resty.Route base;

  final SerializerRepo serializers;

  UserApi({this.base, this.serializers});

  @GetReq("/:id")
  Future<User> getUserById(String id);

  @PostReq("/")
  Future<User> createUser(@AsJson() User user);

  @PutReq("/:id")
  Future<User> updateUser(String id, @AsJson() User user);

  @DeleteReq("/:id")
  Future<void> deleteUser(String id);

  @GetReq("/")
  Future<List<User>> all({String name, String email});
}
```

### PR DESCRIPTION
**Pull request by jaumard**
fix #1

### CODE REVIEW

File: retrofit/lib/annotations/annotations.dart [hunk 1]
```diff
@@ -2,5 +2,6 @@ export 'requests.dart';
 export 'params.dart';
 
 class GenApiClient {
-  const GenApiClient();
+  final String basePath;
```
Comment (tejainece) [inline retrofit/lib/annotations/annotations.dart]: Just `path` should be enough?

File: retrofit/lib/annotations/annotations.dart [hunk 2]
```diff
@@ -2,5 +2,6 @@ export 'requests.dart';
...
+  const GenApiClient([this.basePath = ""]);
```
Comment (tejainece) [inline retrofit/lib/annotations/annotations.dart]: Please make `basePath` named optional argument.

File: retrofit_gen/lib/writer/writer.dart [hunk 3]
```diff
@@ -22,7 +22,7 @@ class Writer {
 
     sb.write('var req = base.${r.method}');
 
-    if (r.path != null) sb.write('.path("${r.path}")');
+    if (r.path != null) sb.write('.path("\$basePath${r.path}")');
```
Comment (tejainece) [inline retrofit_gen/lib/writer/writer.dart]: Use two different calls to `.path` method. That way, Jaguar Resty will take care of missing forward slashes, etc.

**Comment by tejainece** [comment]
@jaumard Please fix the merge conflicts.

File: retrofit/lib/annotations/annotations.dart (hunk 2 above)
Comment (jaumard) [reply retrofit/lib/annotations/annotations.dart]: Do you mean just initialize it to null instead ?

File: retrofit_gen/lib/writer/writer.dart (hunk 3 above)
Comment (jaumard) [reply retrofit_gen/lib/writer/writer.dart]: ho ok !

**Comment by jaumard** [comment]
Let me know if there something more to do @tejainece should be ok now

File: retrofit/lib/annotations/annotations.dart [hunk 4]
```diff
@@ -2,5 +2,6 @@ export 'requests.dart';
 export 'params.dart';
 
 class GenApiClient {
-  const GenApiClient();
+  final String path;
+  const GenApiClient([this.path = null]);
```
Comment (tejainece) [inline retrofit/lib/annotations/annotations.dart]: Please change it to named optional parameter.

```dart
const GenApiClient({this.path});
```

Comment (jaumard) [reply retrofit/lib/annotations/annotations.dart]: Done :)

**Pull request merged**